code/modules/mob/living/carbon/human/shock.dm

```diff
@@ -0,0 +1,75 @@
+/mob/living/carbon/human/var/traumatic_shock = 0
+/mob/living/carbon/human/var/shock_stage = 0
+
+// proc to find out in how much pain the mob is at the moment
+/mob/living/carbon/human/proc/updateshock()
+	traumatic_shock = getOxyLoss() + getToxLoss() + getFireLoss() + getBruteLoss() + getCloneLoss()
+
+	// broken or ripped off organs will add quite a bit of pain
+	for(var/thing in bodyparts)
+		var/obj/item/organ/external/BP = thing
+		if(BP.status & ORGAN_BROKEN && !(BP.status & ORGAN_SPLINTED) || BP.open)
+			traumatic_shock += 15
+
+	if(reagents)
```

Comment:
The reduction in trauma was moved here, to after the organ being broken/open because painkillers should dull that, as well.